### PR TITLE
pkp/pkp-lib#9664 Use jobs chain for usage stats log files processing 

### PR DESCRIPTION
**Pull request by bozana**
s. https://github.com/pkp/pkp-lib/issues/9664

### CODE REVIEW

File: jobs/statistics/ProcessUsageStatsLogFile.php [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+            }
+
+            // Avoid bots.
+            if (Core::isUserAgentBot($entryData->userAgent)) {
+                continue;
+            }
+
+            $this->insertTemporaryUsageStatsData($entryData, $lineNumber);
+        }
+        fclose($fhandle);
```
Comment (jonasraoni) [inline jobs/statistics/ProcessUsageStatsLogFile.php:127]: Not a big deal, but it's safer to use:
```php
fopen();
try{
  ...
} finally {
  fclose();
}
```

You might also use the [SplFileObject](https://www.php.net/splfileobject).

File: jobs/statistics/ProcessUsageStatsLogFile.php [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+        $temporaryItemInvestigationsDao->deleteByLoadId($this->loadId);
+        $temporaryItemRequestsDao->deleteByLoadId($this->loadId);
+    }
+
+    /**
+     * Validate the usage stats log entry
+     *
+     * @throws Exception.
+     */
+    protected function isLogEntryValid(object $entry): void
```
Comment (jonasraoni) [inline jobs/statistics/ProcessUsageStatsLogFile.php:151]: As it's not returning a boolean, I think it's better to rename it to `validateLogEntry()` or something like that.

File: jobs/statistics/ProcessUsageStatsLogFile.php [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        $validFileTypes = [
+            StatisticsHelper::STATISTICS_FILE_TYPE_PDF,
+            StatisticsHelper::STATISTICS_FILE_TYPE_DOC,
+            StatisticsHelper::STATISTICS_FILE_TYPE_HTML,
+            StatisticsHelper::STATISTICS_FILE_TYPE_OTHER,
+        ];
+        if (!empty($entry->fileType) && !in_array($entry->fileType, $validFileTypes)) {
+            throw new Exception(__('admin.job.processLogFile.invalidLogEntry.fileType'));
+        }
+        if (!empty($entry->country) && (!ctype_alpha($entry->country) || !(strlen($entry->country) == 2))) {
```
Comment (jonasraoni) [inline jobs/statistics/ProcessUsageStatsLogFile.php:178]: I think it's clearer `strlen($entry->country) !== 2`

File: jobs/statistics/ProcessUsageStatsLogFile.php [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+            throw new Exception(__('admin.job.processLogFile.invalidLogEntry.country'));
+        }
+        if (!empty($entry->region) && (!ctype_alnum($entry->region) || !(strlen($entry->region) <= 3))) {
```
Comment (jonasraoni) [inline jobs/statistics/ProcessUsageStatsLogFile.php:181]: And here `strlen($entry->region) > 3`

**Review by jonasraoni — APPROVED** [review]

File: jobs/statistics/CompileCounterSubmissionDailyMetrics.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * The load ID = usage stats log file name
+     */
+    protected string $loadId;
+
+    /**
+     * Create a new job instance.
+     */
+    public function __construct(string $loadId)
```
Comment (jonasraoni) [inline jobs/statistics/CompileCounterSubmissionDailyMetrics.php:40]: You can use this new syntax to declare the properties that should be also set at the constructor:
```php
public function __construct(protected string $loadId) {
}
```

File: jobs/statistics/ProcessUsageStatsLogFile.php (hunk 1 above)
Comment (bozana) [reply jobs/statistics/ProcessUsageStatsLogFile.php:127]: I will then use SplFileObject...

**Pull request merged**